### PR TITLE
Update patch state colors and DiffStatBadge contrast

### LINKED ISSUE #662: Fix contrast in diff pill in patch list view
In light mode the red-on-red contrast is too low. We should do it similarly to how it is in the commit diff view.

Screenshot 2023-04-03 at 16 00 53
Screenshot 2023-04-03 at 16 01 05
Screenshot 2023-04-03 at 16 01 08
Screenshot 2023-04-03 at 16 01 19

### PR DESCRIPTION
**Pull request by sebastinez**
See PR title

Closes #662, #667

Bildschirmfoto 2023-04-14 um 06 39 54
Bildschirmfoto 2023-04-14 um 06 39 40
Bildschirmfoto 2023-04-14 um 06 39 35
Bildschirmfoto 2023-04-14 um 06 39 31
Bildschirmfoto 2023-04-14 um 06 39 23
Bildschirmfoto 2023-04-14 um 06 39 21
Bildschirmfoto 2023-04-14 um 06 39 16

### CODE REVIEW

File: src/views/projects/Patch.svelte [hunk 1]
```diff
@@ -159,6 +160,17 @@
     currentRevision.base,
     currentRevision.oid,
   );
+
+  const badgeColor: Variant =
```
Comment (rudolfs) [inline src/views/projects/Patch.svelte:164]: I always find nested ternary conditionals hard to read. Could we convert it to if/else?

**Comment by rudolfs** [comment]
I think we can bump the draft state indicator one level brighter.

This is a bit dark:
Screenshot 2023-04-14 at 12 49 35


```diff
diff --git a/src/views/projects/Patch/PatchTeaser.svelte b/src/views/projects/Patch/PatchTeaser.svelte
index bf62b024..9e171cee 100644
--- a/src/views/projects/Patch/PatchTeaser.svelte
+++ b/src/views/projects/Patch/PatchTeaser.svelte
@@ -93,7 +93,7 @@
     color: var(--color-foreground-6);
   }
   .draft {
-    background-color: var(--color-foreground-3);
+    background-color: var(--color-foreground-4);
   }
   .open {
     background-color: var(--color-positive);
```

File: src/views/projects/Patch.svelte (hunk 1 above)
Comment (sebastinez) [reply src/views/projects/Patch.svelte:164]: Yes that should work too

Comment (sebastinez) [reply src/views/projects/Patch.svelte]: 9111fdd

**Comment by sebastinez** [comment]
> I think we can bump the draft state indicator one level brighter.
> 
> This is a bit dark: Screenshot 2023-04-14 at 12 49 35
> 
> ```diff
> diff --git a/src/views/projects/Patch/PatchTeaser.svelte b/src/views/projects/Patch/PatchTeaser.svelte
> index bf62b024..9e171cee 100644
> --- a/src/views/projects/Patch/PatchTeaser.svelte
> +++ b/src/views/projects/Patch/PatchTeaser.svelte
> @@ -93,7 +93,7 @@
>      color: var(--color-foreground-6);
>    }
>    .draft {
> -    background-color: var(--color-foreground-3);
> +    background-color: var(--color-foreground-4);
>    }
>    .open {
>      background-color: var(--color-positive);
> ```

Done here f04756e

File: src/views/projects/Patch.svelte [hunk 2]
```diff
@@ -161,16 +161,19 @@
...
-    patch.state.status === "draft"
-      ? "foreground"
-      : patch.state.status === "open"
-      ? "positive"
-      : patch.state.status === "archived"
-      ? "caution"
-      : patch.state.status === "merged"
-      ? "primary"
-      : "foreground";
+  function getBadgeColor(status: string): Variant {
```
Comment (rudolfs) [inline src/views/projects/Patch.svelte:164]: We can just call it `badgeColor`.

**Review by rudolfs — APPROVED** [review]

**Pull request merged**